src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue

```diff
@@ -383,17 +462,27 @@ import { useActorStore } from '../../../../../stores/actor.ts'
 import { useChatExtrasStore } from '../../../../../stores/chatExtras.ts'
 import { useIntegrationsStore } from '../../../../../stores/integrations.js'
 import { useReactionsStore } from '../../../../../stores/reactions.js'
+import { useSharedItemsStore } from '../../../../../stores/sharedItems.ts'
 import { generatePublicShareDownloadUrl, generateUserFileUrl } from '../../../../../utils/davUtils.ts'
-import { convertToUnix, formatDateTime } from '../../../../../utils/formattedTime.ts'
+import { convertToUnix, formatDateTime, ONE_DAY_IN_MS } from '../../../../../utils/formattedTime.ts'
+import { getRelativeDay } from '../../../../../utils/formattedTime.ts'
 import { getCustomDateOptions } from '../../../../../utils/getCustomDateOptions.ts'
 import { copyConversationLinkToClipboard } from '../../../../../utils/handleUrl.ts'
 import { parseMentions } from '../../../../../utils/textParse.ts'
 
+const PIN_DURATION_OPTIONS = [
+	{ key: '24_hours', label: t('spreed', '24 hours'), duration: ONE_DAY_IN_MS },
+	{ key: '7_days', label: t('spreed', '7 days'), duration: 7 * ONE_DAY_IN_MS },
+	{ key: '30_days', label: t('spreed', '30 days'), duration: 30 * ONE_DAY_IN_MS },
+	{ key: 'indefinitely', label: t('spreed', 'Until unpin'), duration: 0 },
```

Comment:
Maybe `Permanently` or `Always` would sound better here? I'm also having a hard time translate this one into something meaningful and short
cc @nimishavijay